src/Microsoft.Diagnostics.Monitoring.RestServer/Auth/ApiKeyConfiguration.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    internal sealed class ApiAuthenticationOptions
+    {
+        public const string ConfigurationKey = "ApiAuthentication";
+
+        public string ApiKeyHash { get; set; }
+        public string HashType { get; set; }
```

Comment:
nit: maybe ApiKeyHashType to make it clear this is the type of hash of the api key.